sw/airborne/firmwares/rotorcraft/oneloop/oneloop_andi.c

```diff
@@ -0,0 +1,1524 @@
+    for (i = 0; i < ANDI_NUM_ACT; i++) {
+     a_thrust +=(thrust_cmd_1l - use_increment*actuator_state_1l[i]) * bwls_1l[2][i] / (ratio_u_un[i] * ratio_vn_v[i]);
+    }
+    // Overwrite Yaw Ref with desired Yaw setting (for consistent plotting)
+    att_ref[2] = psi_des_rad;
+    // Set desired Yaw rate with stick input
+    des_r = (float) (radio_control_get(RADIO_YAW))/MAX_PPRZ*3.0; 
+    BoundAbs(des_r,3.0);
+    // Generate reference signals
+    rm_2nd(dt_1l, &att_d_ref[2], &att_2d_ref[2], &att_3d_ref[2], des_r, k_r_rm, k_r_d_rm);
+    rm_3rd(dt_1l, &att_ref[0],   &att_d_ref[0],  &att_2d_ref[0], &att_3d_ref[0], eulers_zxy_des.phi, k_phi_rm, k_p_rm, k_pdot_rm);
+    rm_3rd(dt_1l, &att_ref[1],   &att_d_ref[1],  &att_2d_ref[1], &att_3d_ref[1], eulers_zxy_des.theta, k_theta_rm, k_q_rm, k_qdot_rm);
+  }else{
+    // Make sure X and Y jerk objectives are active
+    Wv_wls[0] = Wv[0];
+    Wv_wls[1] = Wv[1];
+    // Group RM gains in arrays
+    float k1_pos_rm[3] = {k_N_rm,  k_E_rm,  k_D_rm};
+    float k2_pos_rm[3] = {k_vN_rm, k_vE_rm, k_vD_rm};
+    float k3_pos_rm[3] = {k_aN_rm, k_aE_rm, k_aD_rm};
+    float k1_att_rm[3] = {k_phi_rm, k_theta_rm, k_psi_rm};
+    float k2_att_rm[3] = {k_p_rm, k_q_rm, k_psi_d_rm};
+    float k3_att_rm[3] = {k_pdot_rm, k_qdot_rm, k_psi_2d_rm};
+    // First time engaging NAV Mode or Navigation Functions (e.g. Oval) requires some initialization
+    if(check_1st_nav){ 
+      // Mark that NAV Mode has been engaged
+      check_1st_nav  = false;
+      // Reset desired heading to current heading
+      psi_des_rad = eulers_zxy.psi; 
+    }
+    // Register Arttitude Setpoints from previous loop
+    float att_des[3] = {eulers_zxy_des.phi, eulers_zxy_des.theta, psi_des_rad};
+    // Generate Reference signals for positioning using RM
+    if (rm_order_h == 3){
+      rm_3rd_pos(dt_1l, pos_ref, pos_d_ref, pos_2d_ref, pos_3d_ref, nav_target, k1_pos_rm, k2_pos_rm, k3_pos_rm, max_v_nav, max_a_nav, max_j_nav, 2);    
+    } else if (rm_order_h == 2){
+      float_vect_copy(pos_ref, pos_1l,2);
+      rm_2nd_pos(dt_1l, pos_d_ref, pos_2d_ref, pos_3d_ref, nav_target, k2_pos_rm, k3_pos_rm, max_a_nav, max_j_nav, 2);   
+    } else if (rm_order_h == 1){
+      float_vect_copy(pos_ref, pos_1l,2);
+      float_vect_copy(pos_d_ref, pos_d,2);
+      rm_1st_pos(dt_1l, pos_2d_ref, pos_3d_ref, nav_target, k3_pos_rm, max_j_nav, 2);   
+    }
+
+    // The RM functions want an array as input. Create a single entry array and write the vertical guidance entries. 
+    float single_value_ref[1]        = {pos_ref[2]};
+    float single_value_d_ref[1]      = {pos_d_ref[2]};
+    float single_value_2d_ref[1]     = {pos_2d_ref[2]};
+    float single_value_3d_ref[1]     = {pos_3d_ref[2]};
+    float single_value_nav_target[1] = {nav_target[2]};
+    float single_value_k1_rm[1]      = {k1_pos_rm[2]};
+    float single_value_k2_rm[1]      = {k2_pos_rm[2]};
+    float single_value_k3_rm[1]      = {k3_pos_rm[2]};
+
+    if (rm_order_v == 3){
```

Comment:
rm_order_v is not the most descriptive name, and the _v seems superfluous

Reply:
I have added a description of this parameter